File: packages/transform/src/features/protobuf/ProtobufCreateIsEncodeTransformer.ts
import { ProtobufIsEncodeProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufCreateIsEncodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "protobuf.createIsEncode",
      write: ProtobufIsEncodeProgrammer.write,
    });
}

File: packages/transform/src/features/protobuf/ProtobufCreateIsEncodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufCreateIsEncodeTransformer = protobufCreateIsEncodeTransformerNamespace{}

type protobufCreateIsEncodeTransformerNamespace struct{}

func (protobufCreateIsEncodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufIsEncodeProgrammer.Write(call.TypeText)
}
